modular_skyrat/modules/customization/modules/mob/dead/new_player/sprite_accessories/taur_types.dm

```diff
@@ -7,27 +7,98 @@
-	var/hide_markings = FALSE //Any taur part that has "legs" should not hide markings
-
-/datum/sprite_accessory/taur/is_hidden(mob/living/carbon/human/H, obj/item/bodypart/HD)
-	if(H.wear_suit && (H.wear_suit.flags_inv & HIDEJUMPSUIT))
+	organ_type = /obj/item/organ/external/taur_body
+	flags_for_organ = SPRITE_ACCESSORY_HIDE_SHOES
+	/// Must be a single specific tauric suit variation bitflag. Don't do FLAG_1|FLAG_2
+	var/taur_mode = NONE
+	/// Must be a single specific tauric suit variation bitflag. Don't do FLAG_1|FLAG_2
+	var/alt_taur_mode = NONE
+
+/datum/sprite_accessory/taur/is_hidden(mob/living/carbon/human/target, obj/item/bodypart/limb)
+	var/obj/item/clothing/suit/worn_suit = target.wear_suit
+	if(istype(worn_suit) && (worn_suit.flags_inv & HIDEJUMPSUIT) && !worn_suit.gets_cropped_on_taurs)
 		return TRUE
-	if(H.owned_turf)
+	if(target.owned_turf)
 		var/list/used_in_turf = list("tail")
-		if(H.owned_turf.name in used_in_turf)
+		if(target.owned_turf.name in used_in_turf)
 			return TRUE
 	return FALSE
 
+
+/obj/item/organ/external/taur_body
+	name = "taur body"
+	zone = BODY_ZONE_CHEST
+	slot = ORGAN_SLOT_EXTERNAL_TAUR
+	layers = ALL_EXTERNAL_OVERLAYS
+	external_bodytypes = BODYTYPE_TAUR
+	color_source = ORGAN_COLOR_OVERRIDE
+
+	feature_key = "taur"
+	preference = "feature_taur"
+	mutantpart_key = "taur"
+	mutantpart_info = list(MUTANT_INDEX_NAME = "None", MUTANT_INDEX_COLOR_LIST = list("#FFFFFF", "#FFFFFF", "#FFFFFF"))
+
+
+/obj/item/organ/external/taur_body/override_color(rgb_value)
+	if(mutantpart_key)
+		return mutantpart_info[MUTANT_INDEX_COLOR_LIST][1]
+
+	return rgb_value
+
+
+/obj/item/organ/external/taur_body/get_global_feature_list()
+	return GLOB.sprite_accessories["taur"]
+
+
+/obj/item/organ/external/taur_body/Insert(mob/living/carbon/reciever, special, drop_if_replaced)
+	if(sprite_accessory_flags & SPRITE_ACCESSORY_HIDE_SHOES)
+		external_bodytypes |= BODYTYPE_HIDE_SHOES
+
+	var/obj/item/bodypart/l_leg/taur/new_left_leg = new /obj/item/bodypart/l_leg/taur()
+	var/obj/item/bodypart/l_leg/old_left_leg = reciever.get_bodypart(BODY_ZONE_L_LEG)
+	var/obj/item/bodypart/r_leg/taur/new_right_leg = new /obj/item/bodypart/r_leg/taur()
+	var/obj/item/bodypart/r_leg/old_right_leg = reciever.get_bodypart(BODY_ZONE_R_LEG)
+
+	new_left_leg.bodytype |= external_bodytypes
+	new_left_leg.replace_limb(reciever, TRUE)
+	if(old_left_leg)
+		qdel(old_left_leg)
+
+	new_right_leg.bodytype |= external_bodytypes
+	new_right_leg.replace_limb(reciever, TRUE)
+	if(old_right_leg)
+		qdel(old_right_leg)
+
+	return ..()
+
+
+/obj/item/organ/external/taur_body/Remove(mob/living/carbon/organ_owner, special, moving)
+	var/obj/item/bodypart/l_leg/left_leg = organ_owner.get_bodypart(BODY_ZONE_L_LEG)
+	var/obj/item/bodypart/r_leg/right_leg = organ_owner.get_bodypart(BODY_ZONE_R_LEG)
+
+	if(left_leg)
+		left_leg.drop_limb()
+
+		if(left_leg)
+			qdel(left_leg)
+
+	if(right_leg)
+		right_leg.drop_limb()
+
+		if(right_leg)
+			qdel(right_leg)
```

Comment:
```suggestion
	if(left_leg)
		left_leg.drop_limb()
		qdel(left_leg)

	if(right_leg)
		right_leg.drop_limb()
		qdel(right_leg)
```
is there a need to check for each leg twice here? I'm not familiar with `drop_limb()` but it seems unnecessary at a glance

Reply:
It's necessary, because otherwise we could hit a runtime when we change stuff around in the prefs menu, as when `drop_limb()` tries to drop a limb to an invalid loc (turf is null, for instance), it's going to qdel it itself. So yeah it'd likely runtime in the prefs menu otherwise.